jax/_src/numpy/lax_numpy.py

```diff
@@ -7807,9 +7807,45 @@ def _roll_static(a: Array, shift: Sequence[int], axis: Sequence[int]) -> Array:
                         dimension=ax)
   return a
 
-@util.implements(np.roll)
 def roll(a: ArrayLike, shift: ArrayLike | Sequence[int],
          axis: int | Sequence[int] | None = None) -> Array:
+  """Roll the elements of an array along a specified axis.
+
+  JAX implementation of :func:`numpy.roll`.
+
+  Args:
+    a: input array.
+    shift: the number of positions to shift the specified axis. If an integer,
+      all axes are shifted by the same amount. If a tuple, the shift for each
+      axis is specified individually.
+    axis: the axis or axes to roll. If ``None``, the array is flattened, shifted,
+      and then reshaped to its original shape.
+
+  Returns:
+    A copy of ``a`` with elements rolled along the specified axis or axes.
+
+  See also:
+    - :func:`jax.numpy.rollaxis`: roll the specified axis to a given position.
+
+  Examples:
+    >>> a = jnp.array([0, 1, 2, 3, 4, 5])
+    >>> jnp.roll(a, 2)
+    Array([4, 5, 0, 1, 2, 3], dtype=int32)
+
+    Roll elements along a specific axis:
+
+    >>> a = jnp.array([[ 0,  1,  2,  3],
+    ...                [ 4,  5,  6,  7],
+    ...                [ 8,  9, 10, 11]], dtype=int32)
+    >>> jnp.roll(a, 1, axis=0)
+    Array([[ 8,  9, 10, 11],
+           [ 0,  1,  2,  3],
+           [ 4,  5,  6,  7]], dtype=int32)
+    >>> jnp.roll(a, [2, 3], axis=1)
```

Comment:
Remove `axis=1` here? Then it will roll separately along the two dimensions

Reply:
how about `jnp.roll(a, [2, 3], axis=0)` ? since it's one of the examples for 'Roll elements along a specific axis'.

Reply:
What's the intended behavior here? Roll 2 followed by roll 3? So `jnp.roll(a, [2, 3], axis=0)` is equivalent to `jnp.roll(a, 5, axis=0)`? That's really counter-intuitive, and is probably more an implementation detail than and intented use of the API.

I think a more useful example would be to show that you can roll `2` along the first axis, and `3` along the second axis: `jnp.roll(a, (2, 3), axis=(0, 1))`